test/expect/TestJit.test_cpp.expect

```diff
@@ -27,3 +27,25 @@ with GraphExecutor_0 = graph(%1 : UNKNOWN_TYPE
   return (%6);
 }
 
+testDifferentiate
+graph(%0 : Float(2, 3, 4)
+      %1 : Float(2, 3, 4)) {
+  %2 : Float(2, 3, 4) = mul(%0, %1)
+  %3 : Float(2, 3, 4) = mul(%2, %0)
+  %4 : Float(2, 3, 4) = add[alpha={1}](%3, %1)
+  return (%4, %2);
+}
+graph(%0 : Float(2, 3, 4)
+      %1 : Float(2, 3, 4)
+      %2 : Float(2, 3, 4)
```

Comment:
At times like this I really wish we encoded the derivative metadata information into the names themselves; even something as simple as `d0` :)

Reply:
Yes, I've been planning to add this at some point too